topics/w5-programming-by-demonstration.md

```diff
@@ -9,3 +9,25 @@
 #### Optional Reading
 
 * [Flashfill](https://support.microsoft.com/en-us/office/using-flash-fill-in-excel-3f9bcf1e-db93-4890-94a0-1578341f73f7): Gulwani, S. (2011). Automating string processing in spreadsheets using input-output examples. ACM Sigplan Notices, 46(1), 317-330.
+
+
+#### Blog Post (Request Review)
+
+Making an inanimate thing perform actions automatically after demonstrating what you want it to do has been a dream for a very long time. No logic that has to be thoroughly thought out and no knowledge of a programming language, simply demonstrate what you want to get done, and the artificial creation itself figures out what you want from it. As so many things in life, reality is usually disappointing. For a program to understand what you want, you usually need to program each action, define what it needs to do for each edge case and even then it might not work as you want it to. For a program to learn from demonstration adds another layer of complexity on top of it, where it has to for example visually, through a video input, or mechanically, through a movement performed by moving the robot arm, understand what you want and then do the programming itself which then guides the movement later. With the added computation power and layer of programming that needs to be done, it is no wonder that the term "programming by demonstration" has only appeared about 30 to 40 years after the first research which became the basis for artificial intelligence. It is still about 40 years old, quite a bit of progress has already been made and still has a lot more potential for improvement.
+
+
```

Comment:
You may want to divide your blog into smaller sections, e.g., rule-based PBDs, programming synthesis-based PBDs, and LLM-based PBDs

Reply:
You may also want to add a broad discussion on the usage scenario of PBD, e.g., UI automation, automated data analysis, robotics automation, IoT configuration, ...